### PR TITLE
Extend `xportr_write` to accept `metadata` and deprecate `label`

### LINKED ISSUE #179: Site examples not using xportr_df_label()
@bms63 @elimillera i'm wondering why in all the site examples we show `xportr_write("adsl.xpt", label = "Subject-Level Analysis Dataset")`, when the `xportr_df_label()` function exists to write the dataset label out?

Wouldn't these examples be cleaner using the dataset labels direct from metacore$ds_spec rather than needing the user to manually provide these?

### PR DESCRIPTION
**Pull request by vedhav**
### Changes Description 

The `xportr_write()` function follows the same design as the rest of the `xportr_*` functions by asking for the `metadata` argument. Thus we're deprecating the `label` argument from this function.

Setting the dataset label can be done either using `xportr_write()` or `xportr_df_label()`

Here is some code to test the `xportr_write()`
```r
devtools::load_all()

metadata <- data.frame(
  dataset = c("adsl", "adae"),
  label = c("Subject-Level Analysis", "Adverse Events Analysis")
)
adsl <- data.frame(
  USUBJID = c(1001, 1002, 1003),
  SITEID = c(001, 002, 003),
  AGE = c(63, 35, 27),
  SEX = c("M", "F", "M")
)

# Writing with no metadata preset and no metadata passed to `xportr_write()`
adsl |>
  xportr_write(
    path = paste0("nono.xpt"),
    strict_checks = TRUE
  )

# Writing with no metadata preset and passing `label` to `xportr_write()` which will be deprecated
adsl |>
  xportr_write(
    path = paste0("noyes1.xpt"),
    strict_checks = TRUE,
    label = "Subject-Level Analysis"
  )
# Writing with no metadata preset and passing `metadata` to `xportr_write()`
adsl |>
  xportr_write(
    path = paste0("noyes2.xpt"),
    strict_checks = TRUE,
    metadata = metadata
  )

adsl <- xportr_df_label(adsl, metadata)
# Writing with metadata preset and no metadata passed to `xportr_write()`
adsl |>
  xportr_write(
    path = paste0("yesno.xpt"),
    strict_checks = TRUE
  )


# Writing with metadata preset and passing `label` to `xportr_write()` which will be deprecated
adsl |>
  xportr_write(
    path = paste0("yesyes1.xpt"),
    strict_checks = TRUE,
    label = "Subject-Level Analysis"
  )
# Writing with metadata preset and passing `metadata` to `xportr_write()`
adsl |>
  xportr_write(
    path = paste0("yesyes2.xpt"),
    strict_checks = TRUE,
    metadata = metadata
  )

nono <- read_xpt("nono.xpt")
noyes1 <- read_xpt("noyes1.xpt")
noyes2 <- read_xpt("noyes2.xpt")
yesno <- read_xpt("yesno.xpt")
yesyes1 <- read_xpt("yesyes1.xpt")
yesyes2 <- read_xpt("yesyes2.xpt")

attr(nono, "label")
# NULL
attr(noyes1, "label")
# [1] "Subject-Level Analysis"
attr(noyes2, "label")
# [1] "Subject-Level Analysis"
attr(yesno, "label")
# [1] "Subject-Level Analysis"
attr(yesyes1, "label")
# [1] "Subject-Level Analysis"
attr(yesyes2, "label")
# [1] "Subject-Level Analysis"
```

### Task List

- [x] The spirit of xportr is met in your Pull Request
- [x] Place Closes #179 into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Summary of changes filled out in the above Changes Description.  Can be removed or left blank if changes are minor/self-explanatory.
- [x] Check that your Pull Request is targeting the `devel` branch, Pull Requests to `main` should use the [Release Pull Request Template](https://github.com/atorus-research/xportr/tree/94_pr_template/.github/PULL_REQUEST_TEMPLATE)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/).  Use `styler` package and functions to style files accordingly.
- [x] Updated relevant unit tests or have written new unit tests. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Unit-Tests) for conventions used in this package. Still at 100% ;) But need to update the tests.
- [x] Creation/updated relevant roxygen headers and examples. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Roxygen-Headers) for conventions used in this package.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Run `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new/updated functions occur on the "Reference" page.
- [x] Update NEWS.md if the changes pertain to a user-facing function (i.e. it has an @export tag) or documentation aimed at users (rather than developers)
- [x] Address any updates needed for vignettes and/or templates
- [x] Link the issue Development Panel so that it closes after successful merging.
- [x] Fix merge conflicts 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Review by elimillera — APPROVED** [review]
A couple small doc comments but looks great

File: README.Rmd [hunk 1]
```diff
@@ -132,7 +132,7 @@ adsl %>%
   xportr_label(var_spec, "ADSL", verbose = "warn") %>%
   xportr_order(var_spec, "ADSL", verbose = "warn") %>%
   xportr_format(var_spec, "ADSL") %>%
-  xportr_write("adsl.xpt", label = "Subject-Level Analysis Dataset")
+  xportr_write("adsl.xpt")
```
Comment (bms63) [inline README.Rmd:135]: If we don't use the xportr_metadata() how does xportr_Write() get the dataset label in this method?

Comment (bms63) [reply README.Rmd:135]: do we need xportr_df_label() here???

File: tests/testthat/test-write.R [hunk 2]
```diff
@@ -26,7 +61,16 @@ test_that("xportr_write: expect error when invalid multibyte string is passed in
...
   on.exit(unlink(tmpdir))
 
-  expect_error(xportr_write(data_to_save, tmp, label = "Lorizzle ipsizzle dolizzl\xe7 pizzle"))
+  expect_error(
+    xportr_write(
+      data_to_save,
+      tmp,
+      metadata = data.frame(
+        dataset = "data_to_save",
+        label = "Lorizzle ipsizzle dolizzl\xe7 pizzle"
```
Comment (bms63) [inline tests/testthat/test-write.R:70]: best label

**Review by bms63 — COMMENTED** [review]
I think I just have concerns on the documentation when we don't use `xportr_metadata()` to set everything for the dataset label.

File: README.Rmd (hunk 1 above)
Comment (vedhav) [reply README.Rmd:135]: Good catch Ben! Yeah, in this case, the df label would be NULL we have to use the `xportr_df_label`. I will make this change wherever we do this.

**Comment by vedhav** [comment]
Not sure if I went overboard by exporting the `dataset_spec` so asking for a re-review. Perhaps just reading it from the path `system.file(paste0("specs/", "ADaM_admiral_spec.xlsx"), package = "xportr")` could have been okay, like we do in the `README.md` What are your thoughts?

**Pull request merged**